Scripts/Runtime/Entities/TaskSystem/AbstractTaskDriver.cs

```diff
@@ -48,35 +54,70 @@ public abstract class AbstractTaskDriver : AbstractAnvilBase
         /// </summary>
         public AbstractTaskSystem TaskSystem { get; }
         
+        /// <summary>
+        /// Reference to the associated <see cref="World"/>
+        /// </summary>
+        public World World { get; }
```

Comment:
🚲 Can just pull world off the system and remove need to pass in a world?

```suggestion
        public World World 
        { 
           get => TaskSystem.World;
        }
```

Reply:
Sadly no. 

When a TaskDriver is created, it will do a `World.GetOrCreateSystem<TTaskSystem>` to create the System if it doesn't already exist. 

You would need the World to do that. 

Let's assume we can somehow get around that, there's no guarantee that the System has hit it's `OnCreate` yet so the system's World doesn't exist.

I've added a comment to explain why we don't pull it off though.

Reply:
> When a TaskDriver is created, it will do a World.GetOrCreateSystem<TTaskSystem> to create the System if it doesn't already exist.

Not sure I understand. Whatever is creating `AbstractTaskDriver` has reference to a system which it's passing through. That system must have reference to a `World`.

> Let's assume we can somehow get around that, there's no guarantee that the System has hit it's OnCreate yet so the system's World doesn't exist.

Doesn't OnCreate get called before `GetOrCreateSystem<>()` returns?

Reply:
> Not sure I understand. Whatever is creating AbstractTaskDriver has reference to a system which it's passing through. That system must have reference to a World.

A `TaskDriver` can be created anywhere. In a `MonoBehaviour` in managed land, in a regular `SystemBase`, in another `TaskDriver` as a SubTaskDriver or inside a `TaskSystem`.  We need to pass the right `World` to that TaskDriver so it's associated with the right `TaskSystem` (which may not have been created yet so we'll kick that off via `GetOrCreateSystem`).

- `MonoBehaviour` - Probably through player-interaction in which case we'll know what the focused World is.
- `SystemBase` - We just pass our own World.
- Parent `TaskDriver` - We pass through the same World we got.
- `TaskSystem` - Similar to `SystemBase` we just pass our own World.

When a `TaskDriver` is created, it could be at any time, especially for the lower level ones that will be composed like Timers, or MoveTo's. 

The `TaskSystems` that govern those may not yet have been created. So `TaskDrivers` ensure they get linked to the right governing system by calling `World.GetOrCreateSystem`. 

> Doesn't OnCreate get called before GetOrCreateSystem<>() returns?

No, this is the whole problem. It only creates the instance. `OnCreate` gets called later on. It should be called `GetOrConstructSystem` I guess.

Reply:
Discussed on a call and we're pushing the `GetOrCreateSystem()` call down to AbstractTaskDriver so we can be certain the World reference matches the world the system exists in.